docs/src/arrays.md

```diff
@@ -0,0 +1,562 @@
+
+```math
+\begin{align*}
+\dot{\Omega}_{ij}
+    &= \sum_k \left( \dot{A}_{ik} B_{kj} + A_{ik} \dot{B}_{kj} \right)
+        && \text{apply scalar product rule }
+            \frac{d}{dt}(x y) = \frac{dx}{dt} y + x \frac{dy}{dt} \\
+    &= \sum_k \dot{A}_{ik} B_{kj} + \sum_k A_{ik} \dot{B}_{kj}
+        && \text{split sum}
+\end{align*}
+```
+
+But the last expression is just the component form of a sum of matrix products:
+
+```math
+\begin{equation}\label{diffprod}
+\dot{\Omega} = \dot{A} B + A \dot{B}
+\end{equation}
+```
+
+This is the matrix product rule, and we write its `frule` as
+
+```julia
+function frule(
+    (_, ΔA, ΔB),
+    ::typeof(*),
+    A::Matrix{<:RealOrComplex},
+    B::Matrix{<:RealOrComplex},
+)
+    Ω = A * B
+    ∂Ω = ΔA * B + A * ΔB
+    return (Ω, ∂Ω)
+end
+```
+
+### Matrix inversion
+
+```julia
+Ω = inv(A)
+```
+
+$$\Omega = A^{-1}$$
+
+It's easiest to derive this rule from either of the two constraints:
+
+```math
+\begin{align*}
+\Omega A &= A^{-1} ~A = I\\
+A \Omega &= A~ A^{-1} = I,
+\end{align*}
+```
+
+where $I$ is the identity matrix.
+
+We use the matrix product rule to differentiate the first constraint:
+
+$$\dot{\Omega} A + \Omega \dot{A} = 0$$
+
+Then, right-multiply both sides by $A^{-1}$ to isolate $\dot{\Omega}$:
+
+```math
+\begin{align}
+0  &= \dot{\Omega}~ A~ A^{-1} + \Omega ~\dot{A}~ A^{-1} \nonumber\\
+   &= \dot{\Omega}~ I + \Omega ~\dot{A}~ A^{-1}
+       && \text{use } A~ A^{-1} = I \nonumber\\
+   &= \dot{\Omega} + \Omega \dot{A} \Omega
+       && \text{substitute } A^{-1} = \Omega \nonumber\\
+\dot{\Omega}
+   &= -\Omega \dot{A} \Omega
+       && \text{solve for } \dot{\Omega} \label{invdiff}
+\end{align}
+```
+
+We write the `frule` as
+
+```julia
+function frule((_, ΔA), ::typeof(inv), A::Matrix{<:RealOrComplex})
+    Ω = inv(A)
+    ∂Ω = -Ω * ΔA * Ω
+    return (Ω, ∂Ω)
+end
+```
+
+### Other useful identities
+
+These identities are particularly useful:
+
+```math
+\begin{align*}
+\frac{d}{dt} \left( \operatorname{real}(A) \right) &= \operatorname{real}(\dot{A})\\
+\frac{d}{dt} \left( \operatorname{conj}(A) \right) &= \operatorname{conj}(\dot{A})\\
+\frac{d}{dt} \left( A^T \right) &= \dot{A}^T\\
+\frac{d}{dt} \left( A^H \right) &= \dot{A}^H\\
+\frac{d}{dt} \left( \sum_{j}  A_{i \ldots j \ldots k} \right) &=
+    \sum_{j} \dot{A}_{i \ldots j \ldots k},
+\end{align*}
+```
+
+where $\cdot^H = \operatorname{conj}(\cdot^T)$ is the conjugate transpose (the `adjoint` function).
+
+## Reverse-mode rules
+
+### Approach
+
+Reverse-mode rules are a little less intuitive, but we can re-use our pushforwards to simplify their derivation.
+Recall our program:
+
+$$t \mapsto (\ldots, X_m, \ldots) \mapsto \Omega \mapsto s,$$
+
+At any step in the program, if we have intermediates $X_m$, we can write down the derivative $\frac{ds}{dt}$ in terms of the tangents $\dot{X}_m = \frac{dX_m}{dt}$ and adjoints $\overline{X}_m = \frac{\partial s}{\partial X_m}$
```

Comment:
Should `$\overline{X}_m$` be superscript, or do you think that would be too confusing for the targeted audience?

Reply:
I considered it, but yeah, I think it would be too confusing. (this tutorial really should be using Einstein notation, which would really simplify the equations, but I also ruled that out for the same reason).

Reply:
Yeah, seems reasonable. Probably also hard to keep this consistent throughout the whole tutorial. I don't think we can use Einstein notation here anyways, since sometimes twice-occuring indices are not summed over.